src/requests/addresses/models.ts

```diff
@@ -0,0 +1,534 @@
+   * object. Not available for user-role tokens
+   */
+  "internalData"?: any,
+  /**
+   * If true then disables forwarded address (stops forwarding messages)
+   */
+  "forwardedDisabled"?: boolean,
+  /**
+   * Autoreply information
+   */
+  "autoreply"?: DateStringAutoreplyModel;
+}
+
+export interface UpdateAddressInforBodyParametersModel {
+  /**
+   * Identity name
+   */
+  "name"?: string,
+  /**
+   * New address if you want to rename existing address
+   * Only affects normal addresses, special addresses that include * can not be changed
+   */
+  "address"?: string,
+  /**
+   * Indicates if this is the default address for the User
+   */
+  "main": boolean,
+  /**
+   * Optional metadata, must be an object or JSON formatted string
+   */
+  "metaData"?: any,
+  /**
+   * Optional metadata for internal use, must be an object or JSON formatted string of an object
+   * Not available for user-role tokens
+   */
+  "internalData"?: any,
+  /**
+   * A list of tags associated with this address
+   */
+  "tags"?: string[]
+}
+
+export interface GetAddressesInfoResponseModelModel {
+  /**
+   * Indicates successful response
+   */
+  "success": boolean,
+  /**
+   * ID of the Address
+   */
+  "id": string,
+  /**
+   * Identity name
+   */
+  "name": string,
+  /**
+   * E-mail address string
+   */
+  "address": string,
+  /**
+   * List of tags associated with the Address
+   */
+  "tags": string[],
+  /**
+   * Metadata object (if available)
+   */
+  "metaData"?: any,
+  /**
+   * Internal metadata object (if available), not included for user-role requests
+   */
+  "internalData"?: any,
+  /**
+   * Indicates if this is the default address for the User
+   */
+  "main": boolean,
+  /**
+   * <date-time>
+   * Datestring of the time the address was created
+   */
+  "created": string
+}
+
+export interface GetAddressInfoQueryParametersModel {
+  /**
+   * If true then resolves also wildcard addresses
+   */
+  allowWildcard: boolean;
+}
+
+export interface GetAddressInfoResponseModel {
+  /**
+   * Indicates successful response
+   */
+  "success": boolean,
+  /**
+   * ID of the Address
+   */
+  "id": string,
+  /**
+   * E-mail address string
+   */
+  "address": string,
+  /**
+   * Identity name
+   */
+  "name": string,
+  /**
+   * ID of the user if the address belongs to a User
+   */
+  "user": string,
+  /**
+   * List of forwarding targets if this is a Forwarded address
+   */
+  "targets": string[],
+  /**
+   * Account limits and usage
+   */
+  "limits": {
+    /**
+     * Forwarding quota
+     */
+    "forwards": LimitUsageTTLModel
+  },
+  /**
+   * Autoreply information
+   */
+  "autoreply": AutoreplyModel,
+  /**
+   * List of tags associated with the Address
+   */
+  "tags": string[],
+  /**
+   * Datestring of the time the address was created
+   */
+  "created": string
+}
+
+export interface ListRegisteredAddressesQueryParametersModel {
```

Comment:
You can extend with `PageQueryModel` to decrease number of columns that needs to be specified